code/src/reducers/todos.js

```diff
@@ -0,0 +1,26 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+export const todos = createSlice({
+  name: 'todos',
+  initialState: {
+    tasks: []
+  },
+  reducers: {
+    addTask: (state, action) => {
+      state.tasks.push({ id: Date.now(), text: action.payload, isCompleted: false })
+    },
+    removeTask: (state, action) => {
+      state.tasks = state.tasks.filter((task) => task.id !== action.payload)
+    },
+    toggleTaskStatus: (state, action) => {
+      const existingTask = state.tasks.find((task) => task.id === action.payload)
+
+      // If the task is already completed set isCompleted to false, otherwise to true to toggle task status
+      existingTask.isCompleted ? existingTask.isCompleted = false : existingTask.isCompleted = true
```

Comment:
Doing assignment within a ternery like this is a little hard to read, I think. Since the `existingTask.isCompleted` condition returns a boolean, you could use `!` to invert it (so saying 'is not true') and use that directly and make this line shorter:

```js
existingTask.isCompleted = !existingTask.isCompleted
```

Reply:
Thank you! That is much better.